src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java

```diff
@@ -135,259 +148,349 @@
-
-        builtInFunctions.put(new FunctionIdentifier(DISTINCTVALUES, 1), DistinctValuesFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(INDEXOF, 2), IndexOfFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(DEEPEQUAL, 2), DeepEqualFunctionIterator.class);
-
-        builtInFunctions.put(new FunctionIdentifier(ABS, 1), AbsFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(CEILING, 1), CeilingFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(FLOOR, 1), FloorFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ROUND, 1), RoundFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ROUND, 2), RoundFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ROUNDHALFTOEVEN, 1), RoundHalfToEvenFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ROUNDHALFTOEVEN, 2), RoundHalfToEvenFunctionIterator.class);
-
-        builtInFunctions.put(new FunctionIdentifier(PI, 0), PiFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(EXP, 1), ExpFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(EXP10, 1), Exp10FunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(LOG, 1), LogFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(LOG10, 1), Log10FunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(POW, 2), PowFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(SQRT, 1), SqrtFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(SIN, 1), SinFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(COS, 1), CosFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(TAN, 1), TanFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ASIN, 1), ASinFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ACOS, 1), ACosFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ATAN, 1), ATanFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(ATAN2, 2), ATan2FunctionIterator.class);
-
-        builtInFunctions.put(new FunctionIdentifier(SUBSTRING, 2), SubstringFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(SUBSTRING, 3), SubstringFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(SUBSTRING_BEFORE, 2), SubstringBeforeFunctionIterator.class);
-        builtInFunctions.put(new FunctionIdentifier(SUBSTRING_AFTER, 2), SubstringAfterFunctionIterator.class);
-        for (int i = 0; i <= 100; i++) {
-            builtInFunctions.put(new FunctionIdentifier(CONCAT, i), ConcatFunctionIterator.class);
+        builtInFunctions.put(position.getIdentifier(), position);
```

Comment:
This looks good. Yes, I am fine with keeping the put outside the creation functions.